CMakeLists.txt

```diff
@@ -215,6 +215,16 @@ else()
   endif()
 endif()
 
+if(DEFINED REF_BLAS_ROOT)
+  if(WIN32)
```

Comment:
For the first WIN32 condition, I will update that, good catch!

For the second point, about finding BLAS_file in FindCBLAS.cmake, for windows, it finds the .lib file rather than the .dll file, so the find_file call would still be necessary for windows regardless. I thought instead of rearranging steps and using variables from within a different file. I would just call the find_file right above the configuration step where it was used for both windows and linux.

Reply:
Hi,

For .dll and .lib extensions I don't understand why you have to find a library with another extension than the one used by all the other tests (I never used windows btw)
Because FindCBLAS.cmake defines the reference library.

You could find the .dll and .lib in Windows and therefore not need to search the library. Or does this cause problems?

So you could just add the line in FindCBLAS.cmake.

Regards,

Reply:
By the way, this is just a small optimization to limit redundant code and make sure to use the same library for all blas tests.

I think asking the project maintainer before changing your code is not a bad idea.

@mmeterel What do you think ? :p

Reply:
For the .dll vs .lib extensions issue, CMake will find the static library .lib, and not the dynamic library that we need to call in GET_LIB_HANDLE() at runtime. So the find_file call is required for finding the .dll file.

We could add the line in FindCBLAS.cmake, but it would require rearranging the lines within CMAKELISTS.txt to ensure that `add_subdirectory(src bin)` happens after the `include(CMakePackageConfigHelpers)` line. For all the variables to be set before the configure_file step is called.

My thoughts on this were that these find_file lines were a new section of the cmake file, where variables for library locations are created before configuration step. But I understand how this could be done in FindCBLAS.cmake to keep all the CBLAS finding in one location

Reply:
So it's good for me